sections/shared/Layout/SideNav/SideNav.tsx

```diff
@@ -43,15 +44,15 @@ const SideNav: FC<SideNavProps> = ({ isDesktop }) => {
 	const addOptimismNetwork = async () => {
 		try {
 			setNetworkError(null);
-			if (!(window.ethereum && window.ethereum.isMetaMask)) {
+			if (process.browser && !(window.ethereum && window.ethereum.isMetaMask)) {
 				return setNetworkError(t('user-menu.error.please-install-metamask'));
 			}
 
 			// metamask mobile throws if iconUrls is included
 			const { chainId, chainName, rpcUrls, blockExplorerUrls } = getOptimismNetwork({
-				layerOneNetworkId: Number(window.ethereum.chainId),
+				layerOneNetworkId: network?.id.valueOf() || 1,
 			});
-			await window.ethereum.request({
+			await (window.ethereum as any).request({
```

Comment:
do we need `as any` ? I thought `ethereum` type got added in the window object in one of our type file.